src/fides/api/ops/service/privacy_request/request_runner_service.py

```diff
@@ -679,28 +659,40 @@ def _retrieve_child_results(  # pylint: disable=R0911
     return results
 
 
-def needs_consent_email_send(
-    db: Session, user_identity: Dict[str, Any], privacy_request: PrivacyRequest
+def get_consent_email_connection_configs(db: Session) -> Query:
+    """Return enabled consent email connection configs."""
+    return db.query(ConnectionConfig).filter(
+        ConnectionConfig.connection_type.in_(CONSENT_EMAIL_CONNECTOR_TYPES),
+        ConnectionConfig.disabled.is_(False),
+        ConnectionConfig.access == AccessLevel.write,
+    )
+
+
+def get_erasure_email_connection_configs(db: Session) -> Query:
+    """Return enabled erasure email connection configs."""
+    return db.query(ConnectionConfig).filter(
+        ConnectionConfig.connection_type.in_(ERASURE_EMAIL_CONNECTOR_TYPES),
+        ConnectionConfig.disabled.is_(False),
+        ConnectionConfig.access == AccessLevel.write,
+    )
+
+
+def needs_email_send(
+    db: Session, user_identities: Dict[str, Any], privacy_request: PrivacyRequest
 ) -> bool:
     """
-    Returns True if the privacy request fulfills the requirements for consent email send:
-
-    1) Privacy request must have consent preferences saved
-    2) There must be consent email connections configured
-    3) The user must have identity data matching at least one of these connectors
+    Delegates the "needs email" check to each configured email or
+    generic email consent connector. Returns true if at least one
+    connector needs to send an email.
     """
-    if not privacy_request.consent_preferences:
-        return False
-
-    email_consent_connection_configs: Query = get_consent_email_connection_configs(db)
 
-    for connection_config in email_consent_connection_configs:
-        secrets: ExtendedConsentEmailSchema = ExtendedConsentEmailSchema(
-            **connection_config.secrets or {}
-        )
-        filtered_user_identities = filter_user_identities_for_connector(
-            secrets, user_identity
-        )
-        if filtered_user_identities:
+    erasure_email_configs: Query = get_erasure_email_connection_configs(db)
+    consent_email_configs: Query = get_consent_email_connection_configs(db)
+    combined_configs = erasure_email_configs.union_all(consent_email_configs)
+    for connection_config in combined_configs:
+        if get_connector(connection_config).needs_email(  # type: ignore
+            user_identities, privacy_request
+        ):
             return True
+
```

Comment:
Nice work here making more generic for both erasure/consent with the "needs_email" method.